### ISSUE TITLE
Use the output of a layer of a base model as the target output of a second model

### ISSUE THREAD

**Issue by carlosg-m** [keras-team/keras#13414]
I have a pre-trained Keras model with an embedding layer.

The objective is to build an autoencoder that compresses the flatten dense array outputted by the embedding. The dense array is both input and output.

The problem is that I cannot use `.predict` method on the base model because I do not have enough ram to store 19.000+ dims. 

I tried to build a generator that uses the base model to predict each batch individually, but Keras cannot handle two models in the same session.

I also tried to use `model_base.output` inside the `.fit` method of the second model, but it didn't work.

**Example:**

```
def model_base():
     input_emb = layers.Input(shape=(312,))
     embedding = layers.Embedding(23000, 64, 312)(input_emb)
     embedding = layers.Flatten()(embedding)
     return tf.keras.model([input_emb], [embedding])

def model_ae(base_input, base_output):
     encoder = layers.Dense(....)(base_output)
     decoder = layers.Dense(312 * 64)(encoder)
     return tf.keras.model([base_input], [decoder])

nn_base = model_base()
nn_base.trainable = False
nn_base.load_weights('/some_weights.h5', by_name=True)

nn_ae = model_ae(nn_base.input, nn_base.output)
nn_ae.compile(optimizer=Adam(learning_rate=0.001), loss='mse', metrics=['mae']) 
nn_ae.fit(x=..., y=nn_base.output)
```

- So my **X** is an array of integers that is fed into the embedding layer. I've tested it and everything is working fine.
- The problem is **Y**, that should be the dense array outputted and flattened by the embedding layer in the base model. However the second model cannot have access to it from the first model!

How can I achieve this result? Keras does not seem to support this.

**Comment by carlosg-m** [comment]
One possible solution is to create a custom loss function that replaces the `y_true` by a specific layer output. When calling `.fit` just pass a dummy array with the same shape as the output of the layer being used as target. This works very well and saves a lot of memory.

This issue can be closed.

```
def model_ae():
     input_emb = layers.Input(shape=(312,))

     embedding = layers.Embedding(23000, 64, 312)(input_emb)
     embedding = layers.Flatten()(embedding)

     encoder = layers.Dense(....)(embedding)
     (...)
     decoder = layers.Dense(312 * 64)(encoder)

     return tf.keras.model([input_emb], [decoder]), \
            tf.keras.model([input_emb], [embedding]), \
            embedding

def custom_loss(target_layer):
     def mse_loss(y_true, y_pred):
         return K.mean(math_ops.squared_difference(y_pred, target_layer), axis=-1)
     return mse_loss

nn_ae, nn_base, target_layer = model_ae()

nn_base.trainable = False
nn_base.load_weights(...)

nn_ae.compile(optimizer=Adam(), loss=custom_loss(target_layer))
nn_ae.fit(x=SomeIntegerArray, y=np.empty((num_lines, 312*64)))
```
